File: leetcode/0121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let current: number = prices[0];
    let maxProfit: number = 0;

    for (let i = 1; i < prices.length; i++) {
        const temp: number = prices[i] - current;

        if (temp < 0) {
            current = prices[i];
        } else {
            maxProfit = Math.max(maxProfit, temp);
        }
    }

    return maxProfit;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int current = prices[0];
        int maxProfit = 0;

        for(int i = 1; i < prices.length; i++) {
            
            int temp = prices[i] - current;
            if(temp < 0) {
                current = prices[i];
            } 
            else {
                maxProfit = Math.max(maxProfit, temp);
            }
        }
        return maxProfit;
    }
}